examples/trainer_keras.py

```diff
@@ -28,83 +33,124 @@ def ensure_results_dir():
+                plot_keras_model_predictions(model, True, step, x_test[0, :, l], y_test[0, :, l], validation_predictions[0, :, l], axis=l)
 
-    model.save('results/n_beats_model.h5')
+    model.nbeats.save('results/n_beats_model.h5')
 
-    predictions = model.predict(x_train)
-    validation_predictions = model.predict(x_test)
-    # plot_keras_model_predictions(model, False, model.steps, x_train[100, :], y_train[100, :], predictions[100, :])
-    plot_keras_model_predictions(model, True, model.steps, x_test[10, :], y_test[10, :], validation_predictions[10, :])
+    if model.has_exog():
+        predictions = model.predict([x_train, e_train])
+        validation_predictions = model.predict([x_test, e_test])
+    else:
+        predictions = model.predict(x_train)
+        validation_predictions = model.predict(x_test)
 
+    for l in range(model.input_dim):
+        plot_keras_model_predictions(model, False, model.steps, x_train[10, :, l], y_train[10, :, l], predictions[10, :, l], axis=l)
+        plot_keras_model_predictions(model, True, model.steps, x_test[10, :, l], y_test[10, :, l], validation_predictions[10, :, l], axis=l)
     print('smape=', get_metrics(y_test, validation_predictions)[0])
     print('error=', get_metrics(y_test, validation_predictions)[1])
 
 
-def plot_keras_model_predictions(model, is_test, step, backcast, forecast, prediction):
+def plot_keras_model_predictions(model, is_test, step, backcast, forecast, prediction, axis):
     legend = ['backcast', 'forecast', 'predictions of forecast']
     if is_test:
-        title = 'results/test/' + str(step) + '.png'
+        title = 'results/test/' + 'step_' + str(step) + '_axis_' + str(axis) + '.png'
     else:
-        title = 'results/' + str(step) + '.png'
+        title = 'results/' + 'step_' + str(step) + '_axis_' + str(axis) + '.png'
     plt.figure()
     plt.grid(True)
     x_y = np.concatenate([backcast, forecast], axis=-1).flatten()
     plt.plot(list(range(model.backcast_length)), backcast.flatten(), color='b')
     plt.plot(list(range(len(x_y) - model.forecast_length, len(x_y))), forecast.flatten(), color='g')
     plt.plot(list(range(len(x_y) - model.forecast_length, len(x_y))), prediction.flatten(), color='r')
     plt.scatter(range(len(x_y)), x_y.flatten(), color=['b'] * model.backcast_length + ['g'] * model.forecast_length)
-    plt.scatter(list(range(len(x_y) - model.forecast_length, len(x_y))), prediction.flatten(),
-                color=['r'] * model.forecast_length)
+    plt.scatter(list(range(len(x_y) - model.forecast_length, len(x_y))), prediction.flatten(), color=['r'] * model.forecast_length)
     plt.legend(legend)
     plt.savefig(title)
     plt.close()
-
-
+        
+        
 def main():
     args = get_script_arguments()
-    model = NBeatsNet()
+
+    # m4
+    if args.task in ['m4', 'dummy']:
+        model = NBeatsNet(input_dim=1, exo_dim=0, backcast_length=10, forecast_length=1, 
```

Comment:
Ideally exo_dim=0 and input_dim=1 should be the default values and we should not have to specify them.

Reply:
Updated